### PR TITLE
Set metadata for reference sets

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #642

### CODE REVIEW

File: ga4gh/exceptions.py [hunk 1]
```diff
@@ -447,7 +447,18 @@ class MissingReferenceMetadata(MalformedException):
...
     def __init__(self, fileName, key):
         self.message = (
-            "JSON metadata for file {} is missing key {}".format(
+            "JSON reference metadata for file {} is missing key {}".format(
+                fileName, key))
+
+
+class MissingReferenceSetMetadata(MalformedException):
+    """
+    A FASTA file is missing some metadata in the corresponding JSON file.
```
Comment (jeromekelleher) [inline ga4gh/exceptions.py]: This comment is a bit misleading --- this is a directory containing FASTAs, not a single fasta file.

File: scripts/download_data.py [hunk 2]
```diff
@@ -292,8 +300,24 @@ def downloadBams(self):
...
+            "sourceAccessions": [],
+            "sourceUri": 'TODO',
+        }
+        referenceSetMetadataFilename = "{}.json".format(
+            self.referenceSetName)
+        dumpDictToFileAsJson(
+            referenceSetMetadata, referenceSetMetadataFilename)
+        # Download chromosomes
+        referenceSetDirList = [self.referenceSetName]
+        mkdirAndChdirList(referenceSetDirList)
```
Comment (jeromekelleher) [inline scripts/download_data.py]: Bit confusing to make this variable just to wrap something in a list. How about:
```python
mkdirAndChdirList([self.referenceSetName])
```

**Comment by jeromekelleher** [comment]
Looks good @dcolligan, thanks. +1 modulo above suggestions.

**Comment by dcolligan** [comment]
@jeromekelleher updated

**Comment by jeromekelleher** [comment]
Thanks @dcolligan --- I'll merge once CI is finished.

**Pull request merged**